java/semantickernel-api/src/main/java/com/microsoft/semantickernel/semanticfunctions/SemanticFunctionConfig.java

```diff
@@ -6,15 +6,15 @@ public class SemanticFunctionConfig {
     private final PromptTemplateConfig config;
 
     /** Prompt template */
-    private final PromptTemplate template;
+    private final String template;
```

Comment:
Since the PromptTemplate held the PromptTemplateEngine, it complicated our logic when it came to building PromptTemplates as you could not construct them without a reference to the kernel/PromptTemplateEngine that was ultimately going to render them. This change allowed leaving providing of the engine until the time the template is rendered, and the kernel that was doing the rendering could provide the engine. 

I have reverted this